### ISSUE TITLE
Puppeteer options not applied

### ISSUE THREAD

**Issue by actuallymentor** [peterdemartini/html5-to-pdf#39]
Hello Peter,

Thank you for this wonderful package.

AN issue I run into with the 3.x upgrade is that my configurations are not applied. There are no errors, but the page size, margins and title are simply ignored. I'm generating with:

```js
async ( content, meta, output ) => {
	const thepdf = new pdf( {
		inputBody: content,
		outputPath: `${output}/${meta.title}.pdf`,
		include: [ { type: 'css', filePath: meta.csspath } ],
		pdf: {
			title: meta.title,
			format: 'A5',
			margin: {
				top: '20px',
				bottom: '20px',
				right: '50px',
				left: '50px'
			}
		}
	} )
	await thepdf.start()
	await thepdf.build()
	await thepdf.close()
},
```
 Have I misunderstood how the `.pdf` property is used to pass configs to puppeteer?

**Comment by crypton480** [comment]
I'm running into the same issue. I suspect it is due to the `omitBy` function on line 30 in `lib/index.js`. The pdf options are being initialized as an empty object. Replacing this by `pickBy`, I am able to pass some settings. In particular, printBackground, format, pageRanges work well. However, landscape option seems to be ignored here too.

**Comment by schm-dt** [comment]
I believe I am running into the same issue - applying @crypton480 's fix works for me.

**Comment by actuallymentor** [comment]
For complete discussion, this is line 30:

```js
const pdf = omitBy(
   options.pdf || {
    landscape: legacyOptions.landscape,
    format: legacyOptions.pageSize,
    printBackground: legacyOptions.printBackground,
  },
)
```

I'm guessing that the goal of this is to fill `const pdf` with `options.pdf` and to fill `landscape, format, printBackgorund` if they are not found in `options.pdf`. (is this the intention @peterdemartini ?)

The [lodash docs for omitby](https://lodash.com/docs/4.17.10#omitBy) describe the omitby function as:

```js
// The opposite of _.pickBy; this method creates an object composed of the own and inherited enumerable string keyed properties of object that predicate doesn't return truthy for. The predicate is invoked with two arguments: (value, key).

var object = { 'a': 1, 'b': '2', 'c': 3 };
 
_.omitBy(object, _.isNumber);
// => { 'b': '2' }
```

To me it looks like the second argument (the filter function) is missing, unless there is an undocumented behavior to the function when the argument is left empty.

@peterdemartini could you clarify whether you left the second argument empty on purpose?

**Comment by crypton480** [comment]
@actuallymentor Thanks for the detailed post! I believe the second argument being empty implies that it defaults to the `_.identity` function. http://underscorejs.org/#identity
So applying `omitBy` with the identity function as the filter, we will always get an empty object. Kind of how taking an intersection of a set with its complement will always be empty.
```
var object = { 'a': 1, 'b': '2', 'c': 3 }; //This could be any object
_.omitBy(object, _.isIdentity);
// => {}```

**Comment by peterdemartini** [comment]
Awesome! You guys found a pretty serious issue. The default predicate is identity, the second argument isn’t required. The issue is identity is not right. It want to remove values that are existential not falsely. I wanted the equivalent of `value == null` not `!value`.

I’d be more than happy to accept a pull-request for the change, if you are for it? I want to make sure you guys get credit for finding this. If not, I’d can do this.

**Comment by 1000i100** [comment]
Same issue with the omitBy
removing it worked for me, but not in CI, i'm witing for the pull-request ;)

**Comment by crypton480** [comment]
Sorry, for the late reply. I was away this week. I just created the pull-request. Thanks, @peterdemartini! :)

**Comment by peterdemartini** [comment]
I merged the pull-request, and 3.1.1 should be going out soon!
